src/makeInfinityAdder.js

```diff
@@ -5,6 +5,30 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let total = 0;
+  return (a) => {
+    total = a;
+
+    if (a === undefined) {
+      return 0;
```

Comment:
Returning `0` directly here is not correct as it does not account for the accumulated total. Instead, you should return the current value of `total` and then reset it.